### PR TITLE
Scripts

### PR DESCRIPTION
**Pull request by angelinechen9**
I wrote a Node.js script that migrates data from Google Sheets to Cloud Firestore.

### CODE REVIEW

File: migration.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+                        deal["foodbankId"] = doc.id;
+                        db.collection("surplus")
+                          .add(surplus)
+                          .then((doc) => {
+                            deal["surplusId"] = doc.id;
+                            db.collection("deals").add(deal);
+                          });
+                      });
+                  });
+              });
```
Comment (jessimb) [inline migration.js:236]: Should there be any level of error handling here?

File: migration.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+/**
+ * Get and store new token after prompting for user authorization, and then
+ * execute the given callback with the authorized OAuth2 client.
+ * @param {google.auth.OAuth2} oAuth2Client The OAuth2 client to get token for.
+ * @param {getEventsCallback} callback The callback for the authorized client.
+ */
+function getNewToken(oAuth2Client, callback) {
+  const authUrl = oAuth2Client.generateAuthUrl({
+    access_type: "offline",
```
[the remark this answers is not in the document]
Comment (angelinechen9) [reply migration.js:50]: The [quickstart](https://developers.google.com/sheets/api/quickstart/nodejs) set the value to offline.  According to the [documentation](https://developers.google.com/identity/protocols/oauth2/web-server), "set the value to offline if your application needs to refresh access tokens when the user is not present at the browser".

File: migration.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+              columns["$ Shipping"] = i;
+              break;
+            case "$ in Total":
+              columns["$ in Total"] = i;
+              break;
+          }
+        }
+        rows.forEach((row, index) => {
+          if (index != 0) {
+            let produce = {
```
[the remark this answers is not in the document]
Comment (angelinechen9) [reply migration.js:148]: I used `const` because the properties of the objects are changed when the databases are queried.

File: migration.js [hunk 4]
```diff
@@ -86,61 +96,36 @@ function migrateDeals(auth) {
...
+          "Delivery Date": undefined,
+          "Produce Type": undefined,
+          "Produce Quantity": undefined,
+          "$ to Farm": undefined,
+          "$ Shipping": undefined,
+          "$ in Total": undefined,
+        };
+        for (let i = 0; i < headerRow.length; i++) {
+          if (headerRow[i] in columns) {
+            columns[headerRow[i]] = i;
```
Comment (jessimb) [inline migration.js:128]: much cleaner and more maintainable

**Review by jessimb — APPROVED** [review]
LGTM

**Pull request merged**